lib/rpmug.c

```diff
@@ -1,16 +1,79 @@
 #include "system.h"
 
-#include <pthread.h>
-#include <pwd.h>
-#include <grp.h>
-#include <netdb.h>
+#include <errno.h>
 #include <rpm/rpmlog.h>
 #include <rpm/rpmstring.h>
 
 #include "lib/misc.h"
 #include "lib/rpmug.h"
 #include "debug.h"
 
+#define PWFILE "/etc/passwd"
+#define GRPFILE "/etc/group"
```

Comment:
I'm not aware of another system that would put passwd and group somewhere other than /etc, but then I think maybe CoreOS would and I'm sure MacOS X has a place like /private/etc or something like that.  Could these be defined in /usr/lib/rpm/macros or /usr/lib/rpm/rpmrc or is that too much yak shaving for unknown future proofing?  I personally like giving integrators the ability to adjust file locations at runtime if possible, but I know these are special files.

Reply:
I wondered about this too. If there's a remotely relevant OS that has those files in some other location or some other use-case, I'd like to hear. That said, an "obvious" use-case for overrides is testing, and that may well be enough of a reason to do support it.

Oh and thanks for the review!

Reply:
Configuration macros added now.